### ISSUE TITLE
[BUG] `hybrid=False` leads to no trainable params with jax

### ISSUE THREAD

**Issue by albi3ro** [PennyLaneAI/pennylane#7614]
### Expected behavior

I would expect us to fall back to treating all parameters as trainable with `hybrid=False`.

### Actual behavior

/Users/christina/Prog/pennylane/pennylane/gradients/metric_tensor.py:354: UserWarning: Attempted to compute the metric tensor of a tape with no trainable parameters. If this is unintended, please mark trainable parameters in accordance with the chosen auto differentiation framework, or via the 'tape.trainable_params' property.
  warnings.warn(
()

### Source code

```shell
@qml.qnode(qml.device('lightning.qubit', wires=3))
def c(x):
    qml.RX(2*x, 0)
    return qml.expval(qml.Z(0))

qml.metric_tensor(c, hybrid=False)(jax.numpy.array(0.5))
```

### Tracebacks

```shell

```

### System information

```shell
master
```

### Existing GitHub issues

- [x] I have searched existing GitHub issues to make sure the issue does not already exist.
